### PR TITLE
Add image embedding component

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
PR that adds the image embedding component. Largely inspired by Niel's PR #111 (inference and batching with dask).

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
Can you rebase this one @PhilippeMoussalli?

**Comment by PhilippeMoussalli** [comment]
> Can you rebase this one @PhilippeMoussalli?

Done

File: examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+from fondant.component import TransformComponent
+from fondant.logger import configure_logging
+
+configure_logging()
+logger = logging.getLogger(__name__)
+
+
+@dask.delayed
+def load(example):
```
Comment (ChristiaensBert) [inline examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py:24]: doc string?

File: examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    bytes = io.BytesIO(example)
+    image = Image.open(bytes).convert("RGB")
+    return image
+
+
+@dask.delayed
+def transform(image, processor, device):
```
Comment (ChristiaensBert) [inline examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py:31]: doc string?

File: examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        gpu_model = dask.delayed(model.to(device))
+        embeddings = [embed(batch, gpu_model) for batch in batches]
+
+        # join lists into a single Dask delayed object
+        embeddings = flatten(embeddings)
+        delayed_series = dd.from_delayed(embeddings, meta=pd.Series(dtype="object"))
+        embeddings_df = delayed_series.to_frame(name="embeddings_data")
+
+        # add index columns
+        embeddings_df["id"] = dataframe["id"].reset_index(drop=True)
```
Comment (ChristiaensBert) [inline examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py:113]: is it needed to reset_index() separately on all the columns individually?

File: examples/pipelines/finetune_stable_diffusion/components/image_embedding/fondant_component.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+output_subsets:
+  embeddings:
+    fields:
+      data:
+        type: float32_list
+
+args:
+  model_id:
+    description: Model id on the Hugging Face hub
```
Comment (ChristiaensBert) [inline examples/pipelines/finetune_stable_diffusion/components/image_embedding/fondant_component.yaml:19]: Should we put a default/example value here too? Some people might not know where to look to find the correct CLIP Model ID

File: fondant/schema.py [hunk 5]
```diff
@@ -51,7 +51,7 @@ class Type(enum.Enum):
 
     int8_list = pa.list_(pa.int8())
 
-    float16_list = pa.list_(pa.float16())
+    float32_list = pa.list_(pa.float32())
```
Comment (ChristiaensBert) [inline fondant/schema.py:54]: Are we only supporting float32_list then or do we also need float16 in some cases?

File: examples/pipelines/finetune_stable_diffusion/pipeline.py [hunk 6]
```diff
@@ -31,7 +39,8 @@
     },
 )
 
-write_to_hub_op = FondantComponentOp(
+
+write_to_hub_op = ComponentOp(
     component_spec_path="components/write_to_hub/fondant_component.yaml",
     arguments={
         "username": "philippemo",
```
Comment (ChristiaensBert) [inline examples/pipelines/finetune_stable_diffusion/pipeline.py:46]: should this username stay your name? Or something like 'demo-pipeline'?

**Review by ChristiaensBert — APPROVED** [review]
Looks good to me. Small comments, mainly about docstrings and other small stuff

File: examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py (hunk 3 above)
Comment (PhilippeMoussalli) [reply examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py:113]: good point, I think we can do it in one line

Comment (PhilippeMoussalli) [reply examples/pipelines/finetune_stable_diffusion/components/image_embedding/src/main.py:113]: good point I don't think it's needed

**Pull request merged**